### PR TITLE
Added a restriction for unverified users on Discord

### PR DESCRIPTION
**Pull request by Lunasocute**
Enable browser to open the local discord app if installed. 

## Summary by CodeRabbit

- **New Features**
	- Added a restriction for unverified users on Discord, limiting their access to only the "get-verified" channel.
	- Updated the verification success page to check for a local Discord installation and use a custom URL scheme for redirection if available.
	- Enhanced user experience for unverified users by restricting access to specific channels.

### CODE REVIEW

File: frontend/src/pages/VerificationSuccessful.jsx
Comment (ccrawford4) [inline frontend/src/pages/VerificationSuccessful.jsx:1]: How does this look? Do you mind sharing a screen-recording?

File: frontend/src/pages/VerificationSuccessful.jsx [hunk 1]
```diff
@@ -100,4 +107,29 @@ function VerificationSuccessful() {
...
+    
+    iframe.src = 'discord://';
+    document.body.appendChild(iframe);
+
+    setTimeout(() => {
+      document.body.removeChild(iframe);
+      resolve(false);
+    }, 1000);
+  });
+}
```
[the remark this answers is not in the document]
Comment (ccrawford4) [reply frontend/src/pages/VerificationSuccessful.jsx:133]: ^ Thoughts on this Luna?

File: frontend/src/pages/VerificationSuccessful.jsx [hunk 2]
```diff
@@ -100,4 +107,33 @@ function VerificationSuccessful() {
...
+    };
+    
+    iframe.src = "discord://";
+    document.body.appendChild(iframe);
+
+    function cleanup() {
+      document.body.removeChild(iframe);
+    }
+  });
+}
```
[the remark this answers is not in the document]
Comment (ccrawford4) [reply frontend/src/pages/VerificationSuccessful.jsx:137]: @Lunasocute I think this comment ^ pretty much sums it up. Attempting to use the URL can be tricky as some browsers handle these events differently and browser security is robust. However I don't know of a better way. One thing you could do to improve this is to implement the timeout mechanism incase onload nor onerror is triggered.

File: frontend/src/pages/VerificationSuccessful.jsx [hunk 3]
```diff
@@ -42,8 +42,15 @@ function VerificationSuccessful() {
...
-          window.location.replace('https://discord.com/channels/' + server);
+          const isDiscordInstalled = isDiscordAppInstalled();
+          if (isDiscordInstalled) {
+            // Attempt to open Discord with custom URL scheme
+            const discordServerUrl = `discord://discord.com/channels/${server}`;
+            window.location.replace(discordServerUrl);
+          } else {
+            // Fallback to your previous code for redirection
+            window.location.replace('https://discord.com/channels/' + server);
+          }
```
[the remark this answers is not in the document]
Comment (ccrawford4) [reply frontend/src/pages/VerificationSuccessful.jsx:53]: @Lunasocute Some additional things to consider ^

**Comment by kofimokome** [comment]
Please resolve merge conflicts

**Pull request closed without merging**